logger_config.py

```
@@ -0,0 +1,57 @@
+# Centralized logging configuration using dictConfig
+logging.config.dictConfig({
+    "version": 1,
+    "disable_existing_loggers": False,  # Ensure existing loggers are not disabled
+    "handlers": {  # Handlers define where and how logs are output
+        "file": {
+            "class": "logger_config.ImmediateFlushRotatingFileHandler",  # Custom handler class for writing logs to a file
+            "filename": "debug.log",
+            "maxBytes": 100000,              # Maximum size of a log file in bytes before rotation
+            "backupCount": 3,                # Number of backup files to keep
+            "formatter": "simple",           # Refer to the formatter defined below
+            "errors": "replace"
```

Ruff (0.8.0)

Trailing comma missing

Add trailing comma

(COM812)